app/controllers/network_policy_controller.go

```diff
@@ -120,16 +130,12 @@ func (npc *NetworkPolicyController) Run(healthChan chan<- *ControllerHeartbeat,
 		default:
 		}
 
-		if watchers.PodWatcher.HasSynced() && watchers.NetworkPolicyWatcher.HasSynced() {
-			glog.V(1).Info("Performing periodic sync of iptables to reflect network policies")
-			err := npc.Sync()
-			if err != nil {
-				glog.Errorf("Error during periodic sync: " + err.Error())
-			} else {
-				sendHeartBeat(healthChan, "NPC")
-			}
+		glog.V(1).Info("Performing periodic sync of iptables to reflect network policies")
+		err := npc.Sync()
```

Comment:
I think we can technically get rid of this because it should be covered by the resync period we pass into the informers (this was the case previously too)

Reply:
same with service and routes controller

Reply:
Alternatively keep this but set the resync period to 0 for all informers

Reply:
Writing controller that handles all the corner cases is hard :)

Ideally (in the sense of strong consistency) we recive an update and act on it. Additionally informer can be set to a resync period which will take care of any missed updates from API server for what ever reason. 

If we remove periodic sync, in this scenario, there is no control loop to reconcile (eventual consistent) desired state and actual state.

Alternativley if we keep resync period to 0 then if for any reason cache/informer miss an update from API server, then it will have stale data.

AFAIK, earlier design and with your changes, controller (w.r.t resync period, and periodic loop) is similar to kube-proxy.

Reply:
I think the right thing for our case is to set resync period on informers to 0 and keep the main period sync loop we have for each controller (can be in a future PR). Reason being is that if each `OnUpdate` call is just calling `npc.sync()` then we're calling `npc.sync()` per resource update event when we only need to call it once.

Reply:
Right now we set resync period and run a separate syncloop which is is unnecessary.